From fc15147e72367ca18515f34733302d76c9c4920c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Thu, 27 Oct 2016 14:37:03 +0100
Subject: [PATCH] in msword the hard-break between image and caption has a
 width

while for us it doesn't, make it invisible to give it zero width
in both implementations to head off interoperability misery

Change-Id: I0944006817944b20ef35502c8588357e7ee54810
---
 sw/source/core/frmedt/fews.cxx | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/sw/source/core/frmedt/fews.cxx b/sw/source/core/frmedt/fews.cxx
index 7c5043f..66bc169 100644
--- a/sw/source/core/frmedt/fews.cxx
+++ b/sw/source/core/frmedt/fews.cxx
@@ -18,6 +18,7 @@
  */
 
 #include <svx/svdobj.hxx>
+#include <editeng/charhiddenitem.hxx>
 #include <init.hxx>
 #include <fesh.hxx>
 #include <pagefrm.hxx>
@@ -493,8 +494,15 @@ void SwFEShell::InsertLabel( const SwLabelType eType, const OUString &rText, con
 
                 //put a hard-break after the graphic to keep it separated
                 //from the caption text if the outer frame is resized
-                SwIndex aIdx(pTextNode, bBefore ? nInsertPos : 1);
+                const sal_Int32 nIndex = bBefore ? nInsertPos : 1;
+                SwIndex aIdx(pTextNode, nIndex);
                 pTextNode->InsertText("\n", aIdx);
+                //set the hard-break to be hidden, otherwise it has
+                //non-zero width in word and so hard-break flows on
+                //the next line, pushing the caption text out of
+                //the frame making the caption apparently disappear
+                SvxCharHiddenItem aHidden(true, RES_CHRATR_HIDDEN);
+                pTextNode->InsertItem(aHidden, nIndex, nIndex + 1);
             }
         }
 
-- 
2.9.3

